src/main/webapp/app/exam/participate/exercises/modeling/modeling-submission-exam.component.ts

```diff
@@ -0,0 +1,157 @@
+import { stringifyIgnoringFields } from 'app/shared/util/utils';
+import { ExamParticipationService } from 'app/exam/participate/exam-participation.service';
+
+@Component({
+    selector: 'jhi-modeling-submission-exam',
+    templateUrl: './modeling-submission-exam.component.html',
+    styleUrls: ['./modeling-submission-exam.component.scss'],
+})
+export class ModelingSubmissionExamComponent implements OnInit, OnDestroy, ComponentCanDeactivate {
+    @ViewChild(ModelingEditorComponent, { static: false })
+    modelingEditor: ModelingEditorComponent;
+
+    @Input()
+    participationId: number;
+
+    modelingExercise: ModelingExercise;
+    umlModel: UMLModel; // input model for Apollon
+
+    participation: StudentParticipation;
+    submission: ModelingSubmission;
+
+    hasElements = false; // indicates if the current model has at least one element
+    isSaving: boolean;
+    autoSaveInterval: number;
+
+    constructor(private examParticipationService: ExamParticipationService, private jhiAlertService: AlertService) {
+        this.isSaving = false;
+    }
+
+    ngOnInit(): void {
+        // add
+        this.examParticipationService.getLatestSubmissionForParticipation(this.participationId).subscribe(
+            (modelingSubmission) => {
+                this.updateModelingSubmission(modelingSubmission as ModelingSubmission);
+                this.setAutoSaveTimer();
+            },
+            (error) => {
+                this.jhiAlertService.error(error.message, null, undefined);
+            },
+        );
+
+        window.scroll(0, 0);
+    }
+
+    private updateModelingSubmission(modelingSubmission: ModelingSubmission) {
+        if (!modelingSubmission) {
+            this.jhiAlertService.error('artemisApp.apollonDiagram.submission.noSubmission');
+        }
+
+        this.submission = modelingSubmission;
+        this.participation = modelingSubmission.participation as StudentParticipation;
+
+        // reconnect participation <--> submission
+        this.participation.submissions = [<ModelingSubmission>omit(modelingSubmission, 'participation')];
+
+        this.modelingExercise = this.participation.exercise as ModelingExercise;
+        this.modelingExercise.studentParticipations = [this.participation];
+        this.modelingExercise.participationStatus = participationStatus(this.modelingExercise);
+        if (this.modelingExercise.diagramType == null) {
+            this.modelingExercise.diagramType = UMLDiagramType.ClassDiagram;
+        }
+        if (this.submission.model) {
+            this.umlModel = JSON.parse(this.submission.model);
+            this.hasElements = this.umlModel.elements && this.umlModel.elements.length !== 0;
+        }
+    }
+
+    /**
+     * This function sets and starts an auto-save timer that automatically saves changes
+     * to the model after at most 60 seconds.
+     */
+    private setAutoSaveTimer(): void {
```

Comment:
I thought that another component takes care of autosave?

Reply:
I asked the same question and as far as I understood the explanation this is needed as a sync for the modeling editor. Maybe @TobiasPr can clarify in more detail

Reply:
the modeling-submission has the specialty, that we need to ask Apollon every x seconds for the current model and compare it to the last state, because Apollon had not the ability to communicate model changes in the past. Thats the reason why we have the timer here. The synchronization the server is handled in the ExamParticipationService in one place

Reply:
Maybe a explanation of our working methods helps here: our goal was to create a slim component for the exercise type in exam mode and then further adapt that component in an integration branch which brings together all the other components. That means that this component is not 100% complete, but we did some more adaptations in another branch where the dependencies already exist. I would say it's not a clean way to do it, but for sake of velocity we did it like that. Because we have the code changes of this branch also in the integration branch, it would be possible to not use this branch anymore. However, we thought it makes sense to merge these initial branches back to develop to avoid having on big branch which contains all the changes